src/pages/search/Search.tsx

```diff
@@ -29,14 +32,75 @@ const Search = () => {
   );
   return (<Fade>
     <Query query={SEARCH_LISTINGS} variables={{ input, ...LISTING_CARD_IMAGE_DIMENSIONS }}>
-      {({ loading, error, data }) =>
-        loading ? <Loading /> :
-        error ? <h2>Error: {error.message}</h2> :
-        <SearchPage listings={data.searchListings} />
-      }
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+
+        if (error) isBadUserInputError(error) ? <SearchErrorPage /> : <ErrorPage error={error} />;
+        
+        const { searchListings } = data;
+        if (searchListings.length < 1) <EmptySearchPage />;
```

Comment:
Looks like a missing `return` here as well

`http://localhost:4200/work/search?locationQuery=The+Moon` shows a page with an empty map, but when I add a `return` here I get "No listings in this area, please try a different city or region."

(Also, might be out of scope for this issue, but should we have a CTA for "Be the first to create a listing in this area"? Based on activity from the new-hosts channel it seems like searching and then creating a listing is a pretty common use case.)

Reply:
would be a good idea! our bag of nice to haves is growing😂